### PR TITLE
Bugfix 11266

### PR DESCRIPTION
**Pull request by elpaso**
@m-kuhn can you have a look at this? It fixes ticket 11266
The commit comment indicates that unlink does not work, it works indeed but with the selected features, I don't know if this was the originally intended behaviour.

### CODE REVIEW

File: src/core/qgsrelation.cpp [hunk 1]
```diff
@@ -187,18 +187,19 @@ QgsFeatureRequest QgsRelation::getReferencedFeatureRequest( const QgsAttributes&
...
     int referencingIdx = referencingLayer()->fields().indexFromName( fieldPair.referencingField() );
+    Q_UNUSED( referencingIdx );
 
     QgsField referencedField = referencedLayer()->fields().at( referencedIdx );
 
     if ( referencedField.type() == QVariant::String )
     {
       // Use quotes
-      conditions << QString( "\"%1\" = '%2'" ).arg( fieldPair.referencedField(), attributes.at( referencingIdx ).toString() );
+      conditions << QString( "\"%1\" = '%2'" ).arg( fieldPair.referencedField(), attributes.at( referencedIdx ).toString() );
```
Comment (m-kuhn) [inline src/core/qgsrelation.cpp]: That looks wrong to me.

attributes originates from a feature from the referencing side.

File: src/gui/attributetable/qgsfeaturelistview.cpp [hunk 2]
```diff
@@ -338,3 +345,14 @@ void QgsFeatureListView::selectRow( const QModelIndex& index, bool anchor )
 
   mFeatureSelectionModel->selectFeatures( QItemSelection( tl, br ), command );
 }
+
+void QgsFeatureListView::setFeatureSelectionManager( QgsIFeatureSelectionManager* featureSelectionManager )
+{
+  if ( mFeatureSelectionManager )
+    delete mFeatureSelectionManager;
```
Comment (m-kuhn) [inline src/gui/attributetable/qgsfeaturelistview.cpp]: No need for the if, `delete 0;` is a noop

File: src/core/qgsrelation.cpp (hunk 1 above)
Comment (elpaso) [inline src/core/qgsrelation.cpp]: Not in this bug use case. But I'll check the caller to see why the attributes are coming from the wrong side and if is there any alternate function to call instead.

File: src/gui/attributetable/qgsfeaturelistview.cpp (hunk 2 above)
Comment (elpaso) [inline src/gui/attributetable/qgsfeaturelistview.cpp]: You're completely right, I've just copied verbatim from https://github.com/qgis/QGIS/blob/master/src/gui/attributetable/qgsattributetableview.cpp#L124

File: src/core/qgsrelation.cpp [hunk 3]
```diff
@@ -187,6 +187,7 @@ QgsFeatureRequest QgsRelation::getReferencedFeatureRequest( const QgsAttributes&
   {
     int referencedIdx = referencedLayer()->fields().indexFromName( fieldPair.referencedField() );
     int referencingIdx = referencingLayer()->fields().indexFromName( fieldPair.referencingField() );
+    Q_UNUSED( referencingIdx );
```
Comment (m-kuhn) [inline src/core/qgsrelation.cpp]: No longer required

Comment (elpaso) [inline src/core/qgsrelation.cpp]: Done, thanks.

**Pull request merged**